src/mutations/itemCategory.ts

```diff
@@ -0,0 +1,37 @@
+import { QueryClient } from 'react-query';
+import * as Api from '../api';
+import { buildItemCategoryKey, MUTATION_KEYS } from '../config/keys';
+import { deleteItemCategoryRoutine, postItemCategoryRoutine } from '../routines';
+import { QueryClientConfig } from '../types';
+
+export default (queryClient: QueryClient, queryConfig: QueryClientConfig) => {
+  const { notifier } = queryConfig;
+
+  queryClient.setMutationDefaults(MUTATION_KEYS.POST_ITEM_CATEGORY, {
+    mutationFn: (payload) =>
+      Api.postItemCategory(payload, queryConfig).then(() => payload),
+    onSuccess: () => {
+      notifier?.({ type: postItemCategoryRoutine.SUCCESS });
+    },
+    onError: (error) => {
+      notifier?.({ type: postItemCategoryRoutine.FAILURE, payload: { error } });
+    },
+    onSettled: (_data, _error, { itemId }) => {
+      queryClient.invalidateQueries(buildItemCategoryKey(itemId));
```

Comment:
I might have missed it, but are you using `buildItemCategoryKey(id)` in any hook? Because then there's no really use of this invalidation. Plus, is it actually invalidating/updating the `itemCategories` array? I'm not sure of that.

Reply:
This key is used in the hook "useItemCategories", which is retrieving categories for a given item. And yes it is actually used to update the itemCategories (after a category is added/deleted for the item)

Reply:
This key is used in the hook "useItemCategories", which is retrieving categories for a given item. Yes it is actually updating the itemCategories when a category is added/deleted